include/drivers/kscan.h

```diff
@@ -55,11 +89,19 @@ typedef int (*kscan_config_t)(const struct device *dev,
 			      kscan_callback_t callback);
 typedef int (*kscan_disable_callback_t)(const struct device *dev);
 typedef int (*kscan_enable_callback_t)(const struct device *dev);
+typedef int (*kscan_config_matrix_t)(const struct device *dev, struct kscan_matrix_config *config);
+typedef int (*kscan_drive_matrix_column_t)(const struct device *dev, int col);
+typedef int (*kscan_read_matrix_row_t)(const struct device *dev, int *row);
+typedef int (*kscan_resume_matrix_detection_t)(const struct device *dev, bool resume);
 
 __subsystem struct kscan_driver_api {
 	kscan_config_t config;
 	kscan_disable_callback_t disable_callback;
 	kscan_enable_callback_t enable_callback;
+	kscan_config_matrix_t config_matrix;
+	kscan_drive_matrix_column_t drive_matrix_column;
+	kscan_read_matrix_row_t read_matrix_row;
+	kscan_resume_matrix_detection_t resume_matrix_detection;
```

Comment:
This should be guarded with CONFIG_KCAN_MATRIX.

Reply:
Done.